EKF/terrain_estimator.cpp

```diff
@@ -139,6 +139,11 @@ void Ekf::fuseHagl()
 			_time_last_hagl_fuse = _time_last_imu;
 			_innov_check_fail_status.flags.reject_hagl = false;
 
+			//constrain _terrain_vpos to be a minimum of _params.rng_gnd_clearance larger than _state.pos(2)
```

Comment:
Looking at your test data, this check should be moved to runTerrainEstimator  at the end of the else section after line 104.